### PR TITLE
LGA 3003: Callback capping improvements

### PR DESCRIPTION
**Pull request by BenMillar-CGI**
## What does this pull request do?

- Time slots are now in descending date order so the latest slots are nearest the top.

- The CSV uploader now accepts "900", for "0900" and "930" for "0930".
This resolves a potential issue where Excel removes the leading zeroes from cells it interprets as numbers.

## Any other changes that would benefit highlighting?

Intentionally left blank.

## Checklist

- [x] Provided JIRA ticket number in the title, e.g. "LGA-152: Sample title"

### CODE REVIEW

**Review by TawneeOwl — DISMISSED** [review]
LGTM

**Review by TawneeOwl — DISMISSED** [review]
LGTM

**Review by Pheonnexx — APPROVED** [review]

File: cla_backend/apps/checker/admin.py [hunk 1]
```diff
@@ -6,6 +6,7 @@
 from django.http.response import HttpResponseRedirect
 from checker.models import ReasonForContacting, CallbackTimeSlot
 from checker.forms import CallbackTimeSlotCSVUploadForm
+import datetime as dt
```
Comment (said-moj) [inline cla_backend/apps/checker/admin.py:9]: Why is datetime being aliased as dt here?
I think using the name of the package when referring to the package is easier to read unless you are importing another datetime package in which aliasing will make sense as there will be a name collision.

File: cla_backend/apps/checker/utils.py [hunk 2]
```diff
@@ -53,8 +53,12 @@ def validate_row(cls, row):
         except Exception:
             raise ValidationError(message=dict(date="Write the date in this format: dd/mm/yyyy"))
         try:
-            assert row[CSV_COL_TIME].strip() in CALLBACK_TIME_SLOTS
-            row[CSV_COL_TIME] = row[CSV_COL_TIME].strip()
+            time = row[CSV_COL_TIME].strip()
+            # Excel will sometimes remove the leading zero from values it interprets as numbers.
+            if time in ["900", "930"]:
```
Comment (said-moj) [inline cla_backend/apps/checker/utils.py:58]: I think if it does not start with 0,1 or 2 then you can safely add the leading 0

File: cla_backend/apps/checker/views.py [hunk 3]
```diff
@@ -155,6 +157,7 @@ def perform_update(self, serializer):
 
 
 class CallbackTimeSlotViewSet(PublicAPIViewSetMixin, APIView):
+    @method_decorator(cache_page(10))  # This is an expensive call
```
[the remark this answers is not in the document]
Comment (TawneeOwl) [reply cla_backend/apps/checker/views.py:160]: This was part of the rebase

**Review by said-moj — DISMISSED** [review]

File: cla_backend/apps/checker/admin.py [hunk 4]
```diff
@@ -19,7 +20,8 @@ class CallbackTimeSlotAdmin(admin.ModelAdmin):
     change_list_template = "admin/checker/callback-time-slots/custom_change_list.html"
     list_display = ("date", "time", "capacity", "remaining_capacity")
     date_hierarchy = "date"
-    ordering = ("date", "time")
+    ordering = ("-date", "time")
```
Comment (BenMillar-CGI) [inline cla_backend/apps/checker/admin.py:23]: As we are now only showing the next 7 days, by default, the date should be in ascending order.

This line should read ‘ordering = (“date”, “time”)’

Many thanks!

**Review by Pheonnexx — APPROVED** [review]

**Review by said-moj — APPROVED** [review]

**Pull request merged**